### DISCUSSION TITLE
setting resolve fallbacks in setWebpackConfig is ignored

### DISCUSSION THREAD

**Discussion by wyqydsyq** [gatsbyjs/gatsby#31493] [Help]
## Description

I'm trying to add `.mdx` support to my netlify cms editor with `netlify-cms-widget-mdx`, which uses  `mdx-scoped-runtime` and `@mdx-js/runtime` under the hood, but I keep getting Webpack errors about `can't resolve 'path'` despite having configured it to have a resolve fallback via `setWebpackConfig`, errors are like:

```
"gatsby-plugin-netlify-cms" threw an error while running the onCreateWebpackConfig lifecycle:

Module not found: Error: Can't resolve 'path' in '/projects/static-site/node_modules/mdx-scoped-runtime/node_modules/vfile'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
	- install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "path": false }
```

[According to the docs](https://www.gatsbyjs.com/docs/how-to/local-development/troubleshooting-common-errors/#issues-with-fs-resolution) this scenario can be fixed by setting up exclude/alias/fallback for the offending packages in the Webpack config, however I have done this and still get the same errors.

### Steps to reproduce

`cms.js`:
```
import CMS, {init} from 'netlify-cms-app';
import {MdxControl, MdxPreview} from 'netlify-cms-widget-mdx';

CMS.registerWidget('mdx', MdxControl, MdxPreview);
init();
```

`gatsby-node.js`:
```
exports.onCreateWebpackConfig = ({actions}) => {
  actions.setWebpackConfig({
    resolve: {
      fallback: {
        fs: false,
        path: false,
      },
    },
  });
};
```

In `gatsby-config.js`:
```
  {
      resolve: 'gatsby-plugin-netlify-cms',
      options: {
        publicPath: 'netlify/admin',
        modulePath: `${__dirname}/cms.js`,
        manualInit: true,
      },
    },
```

### Expected result

Webpack uses resolve fallbacks for `fs` and `path` instead of failing to compile

### Actual result

Webpack gives errors about not being able to resolve `path` even though it has a resolve fallback specified

### Environment

  System:
    OS: Linux 5.11 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 3.1.0 - /usr/bin/fish
  Binaries:
    Node: 13.14.0 - ~/.nvm/versions/node/v13.14.0/bin/node
    npm: 6.14.4 - ~/.nvm/versions/node/v13.14.0/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.96
    Firefox: 88.0.1
  npmPackages:
    gatsby: ^3.2.1 => 3.5.0
    gatsby-plugin-manifest: ^3.3.0 => 3.5.0
    gatsby-plugin-mdx: ^2.3.0 => 2.5.0
    gatsby-plugin-netlify-cms: ^5.6.0-next.1 => 5.6.0-next.1
    gatsby-plugin-page-creator: ^3.3.0 => 3.5.0
    gatsby-plugin-react-helmet: ^4.3.0 => 4.5.0
    gatsby-plugin-react-svg: ^3.0.1 => 3.0.1
    gatsby-plugin-robots-txt: ^1.5.6 => 1.6.2
    gatsby-plugin-sitemap: ^3.3.0 => 3.3.0
    gatsby-plugin-styled-components: ^4.3.0 => 4.5.0
    gatsby-remark-autolink-headers: ^4.1.0 => 4.2.0
    gatsby-remark-highlight-code: ^2.2.1 => 2.2.1
    gatsby-source-filesystem: ^3.3.0 => 3.5.0
    gatsby-transformer-remark: ^4.1.0 => 4.2.0
  npmGlobalPackages:
    gatsby: 3.5.0

**Comment by ryanerringtonatom** [comment]
I'm facing this exact same issue, I was able to resolve the build errors by adding the following config in my `gatsby-node.js` file:

```
{
  resolve: "gatsby-plugin-netlify-cms",
    options: {
      modulePath: `${__dirname}/src/cms/cms.js`,
        customizeWebpackConfig: config => {
          config.resolve = {
            fallback: {
              path: false,
            }
          };
        },
    },
}
```

But i'm now facing a runtime error within Netlify CMS following this "ReferenceError: process is not defined" when trying to edit a MDX based blog.

**Comment by konsalex** [comment]
Any update on this issue?

**Comment by walker-tx** [comment]
Howdy - also facing the same issue here.

**Comment by usernamenumber** [comment]
Also having this error. Why was this converted from [a bug](https://github.com/gatsbyjs/gatsby/issues/31475) to a "discussion" (and then not discussed by any of the devs)? 

I was able to address the issue with no subsequent errors using a variation on @ryanerringtonatom's fix above:

```js
    {
      resolve: "gatsby-plugin-netlify-cms",
      options: {
        modulePath: `${__dirname}/src/cms/cms.ts`,
        customizeWebpackConfig: (config) => {
          config.resolve = {
            ...(config.resolve || {}),
            fallback: {
              ...(config.resolve.fallback || {}),
              path: false,
            },
          };
        },
      },
    },
```

**Comment by zackhall** [comment]
I cannot tell you why. But I had to apply the following two changes to resolve the issues my repo experienced. 

In `gatsby-config.js`:

```js
     ...
    {
      resolve: 'gatsby-plugin-netlify-cms',
      options: {
        modulePath: `${__dirname}/src/cms/cms.js`,
        customizeWebpackConfig: (config) => {
          config.resolve = {
            ...(config.resolve || {}),
            fallback: {
              ...(config.resolve.fallback || {}),
              fs: false,
              path: require.resolve('path-browserify'),
            },
          };
        },
      },
    },
    ...
```

And in `gatsby-node.js`:

```js
exports.onCreateWebpackConfig = ({ actions }) => {
  actions.setWebpackConfig({
    resolve: {
      fallback: {
        fs: false,
        path: require.resolve('path-browserify'),
      }
    }
  })
}
```

At least in my repo, adding one or the other does not seem sufficient. I needed to add both to prevent build errors. 🤷
